agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -91,24 +119,50 @@ class AtumContext private[agent] (
     this
   }
 
+  /**
+   * Adds additional data to the AtumContext.
+   *
+   * @param key   the key of the additional data
+   * @param value the value of the additional data
+   */
   def addAdditionalData(key: String, value: String): Unit = {
     additionalData += (key -> Some(value))
   }
 
+  /**
+   * Returns the current additional data in the AtumContext.
+   *
+   * @return the current additional data
+   */
   def currentAdditionalData: Map[String, Option[String]] = {
     this.additionalData
   }
 
+  /**
+   * Adds a measure to the AtumContext.
+   *
+   * @param measure the measure to be added
+   */
   def addMeasure(newMeasure: Measure): AtumContext = {
     measures = measures + newMeasure
     this
   }
 
+  /**
+   * Adds multiple measures to the AtumContext.
+   *
+   * @param measures the sequence of measures to be added
```

Comment:
```suggestion
   * @param measures set of measures to be added
```

probably just for better accuracy